getinp.f90

```diff
@@ -609,23 +609,39 @@ subroutine getinp()
     if(keyword(iline,1).eq.'over') then
       irest = irest + 1
       irestline(irest) = iline
-      ityperest(irest) = 10
       read(keyword(iline,3),*,iostat=ioerr) restpars(irest,1)
       read(keyword(iline,4),*,iostat=ioerr) restpars(irest,2)
       read(keyword(iline,5),*,iostat=ioerr) restpars(irest,3)
       read(keyword(iline,6),*,iostat=ioerr) restpars(irest,4)
-      if(keyword(iline,2).ne.'plane') ioerr = 1
```

Comment:
This will print an error if the keyword is not a plane. Must be modified if there is another option.

Reply:
Should be
if(keyword(iline,2) .eq. 'plane') then
  ityperest(irest) = 10
else if (keyword(iline,2) .eq. 'xygauss') then
   ityperest(irest) = 14
   ...
else
  ioerr = 1
end if

Reply:
Thanks for the suggestions. Ok, now the initial suggestion looks good too! :D  With chkgrad I am getting the following:
```
  Computing the          218 th of       346272  components. Worst error:    1.4853370650395762E-005
  Computing the          232 th of       346272  components. Worst error:    1.4853370650395762E-005
  Computing the          248 th of       346272  components. Worst error:    1.4853370650395762E-005
  Computing the          264 th of       346272  components. Worst error:    1.4853370650395762E-005
  Computing the          280 th of       346272  components. Worst error:    1.4853370650395762E-005
  Computing the          286 th of       346272  components. Worst error:    1.4853370650395762E-005
  Computing the          302 th of       346272  components. Worst error:    2.5085898342852104E-005
  Computing the          318 th of       346272  components. Worst error:    2.5085898342852104E-005
```
Do you have a range for this? 
Regarding the if statement, that is what I have, but in a different order:
```
      if(keyword(iline,2).eq.'xygauss') then
        ityperest(irest) = 14
        read(keyword(iline,7),*,iostat=ioerr) restpars(irest,5)
        read(keyword(iline,8),*,iostat=ioerr) restpars(irest,6)
      else if (keyword(iline,2).ne.'plane') then
        ioerr = 1
      else
        ityperest(irest) = 10
```
I can swap it if you prefer. I will make a new commit soon.

Reply:
It is not the same, I do not ask if .ne. 'plane' anymore, because since there are many options now, we need to test each one, and by exclusion report the error. Note that all questions are ".eq." in my suggestion.

Reply:
Concerning the gradient. Wait until it is finished, it takes a while. Although perhaps it is a good idea to try with smaller systems. At the end it reports the worst error of all, if it is not greater than 1e-3 it is probably fine.

Reply:
Ok, so the changes are added in the last commit. I essentially took the plane def and modded it to the surface I want. [Here](https://uni-duesseldorf.sciebo.de/s/qJWKkvQXD8bkRag) you can find the chgrad.log. The error stays between 1E-5 and 1E-7 most of the time, but there is a fair amount of outliers. Is it ok?

Reply:
No, not really. What you should check if the maximum error. 0.6 it too high. Most of the errors will be small, because for most of the molecules the distance to the new restriction is such that the function and gradient are zero. There is very likely something wrong with the formula of the gradient or of the function.

Reply:
Ok, I'll take a look. I'll get back to you after some administrative work days. Thanks for the guidelines!

Reply:
Well, I had a hunch I forgot something and seems to work. The link has the updated chkgrad.log.